src/imp.rs

```diff
@@ -468,6 +475,83 @@ where
             .map_err(Error::from_prover_err)
     }
 
+    /// Withdraw the reward a key has reward if accumulated by staking and
+    /// taking part in operating the network.
+    pub fn withdraw<Rng: RngCore + CryptoRng>(
+        &self,
+        rng: &mut Rng,
+        sender_index: u64,
+        staker_index: u64,
+        refund: &PublicSpendKey,
+        gas_limit: u64,
+        gas_price: u64,
+    ) -> Result<Transaction, Error<S, SC, PC>> {
+        let sender = self
+            .store
+            .retrieve_ssk(sender_index)
+            .map_err(Error::from_store_err)?;
+        let sender_psk = sender.public_spend_key();
+
+        let sk = self
+            .store
+            .retrieve_sk(staker_index)
+            .map_err(Error::from_store_err)?;
+        let pk = PublicKey::from(&sk);
+
+        let (inputs, outputs) = self.inputs_and_change_output(
+            rng,
+            &sender,
+            refund,
+            gas_limit * gas_price,
+        )?;
+
+        let stake =
+            self.state.fetch_stake(&pk).map_err(Error::from_state_err)?;
+        if stake.reward == 0 {
+            return Err(Error::NoReward((pk, stake)));
```

Comment:
Is this really an error? Looks like it should return a `None` for that case, instead. We are mixing internal storage errors with logical errors.

Reply:
Given these changes https://github.com/dusk-network/rusk/pull/698, if a "withdraw transaction" is sent and the key has no reward, the contract will panic. I would say it is indeed an error - it will never work. It's technically "possible" to send such a transaction, but it will *always* end in wasted fees.